addons/l10n_dk_nemhandel/wizard/account_move_send.py

```diff
@@ -0,0 +1,195 @@
+            # show nemhandel option if either the ubl option is available or any move already has a ubl file generated
+            # and moves are not processing/done and if partners have an edi format set to one that works for nemhandel
+            invalid_partners = wizard.move_ids.partner_id.commercial_partner_id.filtered(lambda partner: partner.ubl_cii_format != 'oioubl_21')
+            wizard.enable_nemhandel = (
+                wizard.company_id.l10n_dk_nemhandel_proxy_state == 'receiver'
+                and (
+                    wizard.enable_ubl_cii_xml
+                    or any(m.ubl_cii_xml_id and m.nemhandel_move_state not in {'processing', 'done'} for m in wizard.move_ids)
+                )
+                and not invalid_partners
+            )
+
+    @api.depends('company_id.account_edi_proxy_client_ids.edi_mode')
+    def _compute_nemhandel_mode_info(self):
+        mode_strings = {
+            'test': _('Test'),
+            'demo': _('Demo'),
+        }
+        for wizard in self:
+            edi_user = wizard.company_id.account_edi_proxy_client_ids.filtered(
+                lambda usr: usr.proxy_type == 'nemhandel'
+            )
+            mode = mode_strings.get(edi_user.edi_mode)
+            wizard.nemhandel_mode_info = f' ({mode})' if mode else ''
+
+    # -------------------------------------------------------------------------
+    # ATTACHMENTS
+    # -------------------------------------------------------------------------
+
+    def _needs_ubl_cii_placeholder(self):
+        return super()._needs_ubl_cii_placeholder() and not self.checkbox_send_nemhandel
+
+    # -------------------------------------------------------------------------
+    # BUSINESS ACTIONS
+    # -------------------------------------------------------------------------
+
+    def action_send_and_print(self, force_synchronous=False, allow_fallback_pdf=False, **kwargs):
+        # Extends 'account' to force ubl xml checkbox if sending via peppol
+        self.ensure_one()
+
+        if all([self.checkbox_send_nemhandel, self.enable_nemhandel, self.enable_ubl_cii_xml, not self.checkbox_ubl_cii_xml]):
+            self.checkbox_ubl_cii_xml = True
+        if self.checkbox_send_nemhandel and self.enable_nemhandel:
+            for move in self.move_ids:
+                if not move.nemhandel_move_state or move.nemhandel_move_state == 'ready':
+                    move.nemhandel_move_state = 'to_send'
+
+        return super().action_send_and_print(force_synchronous=force_synchronous, allow_fallback_pdf=allow_fallback_pdf, **kwargs)
+
+    @api.model
+    def _call_web_service_after_invoice_pdf_render(self, invoices_data):
+        # Overrides 'account'
+        super()._call_web_service_after_invoice_pdf_render(invoices_data)
+
+        params = {'documents': []}
+        invoices_data_nemhandel = {}
+        for invoice, invoice_data in invoices_data.items():
+            if invoice_data.get('send_nemhandel'):
```

Comment:
nitpick but possible early continue here, the function seems veryyyy long maybe we could split into multiple